sdk/scheduler/src/main/java/com/mesosphere/sdk/http/endpoints/HealthResource.java

```diff
@@ -1,79 +1,582 @@
 package com.mesosphere.sdk.http.endpoints;
 
 import com.mesosphere.sdk.http.ResponseUtils;
-import com.mesosphere.sdk.scheduler.SchedulerConfig;
+import com.mesosphere.sdk.scheduler.plan.Plan;
 import com.mesosphere.sdk.scheduler.plan.PlanCoordinator;
 import com.mesosphere.sdk.scheduler.plan.PlanManager;
+import com.mesosphere.sdk.state.FrameworkStore;
+import com.mesosphere.sdk.state.StateStoreException;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.mesos.Protos;
+import org.json.JSONArray;
+import org.json.JSONObject;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
 
-import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+// CHECKSTYLE:OFF LineLengthCheck
+
 /**
- * A read-only API for checking whether the service is healthy.
+ * This class returns a single code representing the status of the service
+ * with the following priority.
+ * HTTP Response Code, Priority, Reason
+ * 318, 1, Initializing
+ * 200, 1, Running
+ * 500, 1, Error Creating Service
+ * 204, 2, Deploying:Pending
+ * 202, 2, Deploying:Starting
+ * 203, 2, Deploying:Waiting_User
+ * 206, 3, Degraded
+ * 203, 4, Recovering:Pending
+ * 205, 4, Recovering:Starting
+ * 320, 5, Backing Up
+ * 321, 5, Restoring
+ * 326, 6, Upgrade/Rollback/Downgrade
+ * 503,  , Service Unavailable (Priority Undefined)
  */
 @Path("/v1/health")
```

Comment:
Have you tested this in clusters going back to 1.10? (Not sure if mesos treats the codes differently across DC/OS versions)

Reply:
Running tests on a 1.10 CCM cluster now to verify.

Reply:
Ran hello-world sanity on CCM 1.10.9 cluster, other than tests which involve setup via SSH keys (test_volumes, test_zzzrecovery_py), tests passed.